src/inner-slider.js

```diff
@@ -46,10 +46,13 @@ export class InnerSlider extends React.Component {
   trackRefHandler = ref => (this.track = ref);
   adaptHeight = () => {
     if (this.props.adaptiveHeight && this.list) {
-      const elem = this.list.querySelector(
-        `[data-index="${this.state.currentSlide}"]`
-      );
-      this.list.style.height = getHeight(elem) + "px";
+      let maxHeight = 0;
+      for (let i = 0; i < this.props.slidesToShow; i++) {
+        const slideIndex = this.state.currentSlide + i;
+        const elem = this.list.querySelector(`[data-index="${slideIndex}"]`);
+        maxHeight = Math.max(getHeight(elem), maxHeight);
```

Comment:
@Avizura Can you provide an example. Here is mine example:

1. Clone repo `git clone https://github.com/Poky85/react-slick.git`
2. Branch `git checkout adaptive-height-fix`
3. Install and run examples `npm install && npm start`
4. Open `http://localhost:8080/` and see "Adaptive height" example.
5. Slides has various height. Carousel's height adapts to the heighest slide visible.

Without my fix, this doesn't work, because carousel's height is calculated as the height of first visible slide.

I tried your solution, but it doesn't worked in "Adaptive height" example. Also it doesn't make sense to me. You make sum of visible slides heights. But they are positioned side-by-side.